### ISSUE TITLE
[@SerenityWithCucumber] Report Portal fetches results only of the first example of a Scenario Outline. 

### ISSUE THREAD

**Issue by nstbushneva** [Invictum/serenity-reportportal-integration#60]
While publishing results of CucumberWithSerenity tests to Report Portal we observed the issue with [Scenario Outlines](https://www.baeldung.com/cucumber-scenario-outline). It seems Report Portal fetches results only of the first example from a Scenario Outline.
As a result, it doesn't fail the launch even the actual build is failed by unsuccessful examples. 

Dependencies: 
* serenity-reportportal-integration [1.3.0]
* maven-failsafe-plugin [2.22.1]
* serenity-core [2.0.20]
* serenity-cucumber [1.9.21]

The project to reproduce the issue: 
[serenity-with-cucumber-scenario-outlines.zip](https://github.com/Invictum/serenity-reportportal-integration/files/2936669/serenity-with-cucumber-scenario-outlines.zip)

**Comment by Invictum** [comment]
@nstbushneva 
This one will be tricky. Actually for now parametrized tests with data provider are not supported at all. I am planning to add support, but not sure it will be quick.

**Comment by Invictum** [comment]
@nstbushneva 

Fix is available in `1.4.0-SNAPSHOT`, work is still in progress, but I assume it will be connected to small refactor and documentation updates.

**Comment by nstbushneva** [comment]
Hi @Invictum, thank you for your fix. 
It seems something goes wrong for Scenario Outline feature with 
```java
<serenity.reportportal.version>1.4.0-SNAPSHOT</serenity.reportportal.version>
```
I am getting CHILD_START_TIME_EARLIER_THAN_PARENT error: 

```java
com.epam.reportportal.exception.ReportPortalException: Report Portal returned error
Status code: 406
Status message: Not Acceptable
Error Message: Start time of child ['Mon Mar 25 08:03:03 UTC 2019'] item should be same or later than start time ['Mon Mar 25 08:03:04 UTC 2019'] of the parent item/launch '5c988b39af66c600012a0f39'
Error Type: CHILD_START_TIME_EARLIER_THAN_PARENT

        at com.epam.reportportal.service.ReportPortalErrorHandler.handleError(ReportPortalErrorHandler.java:83)
        at com.epam.reportportal.service.ReportPortalErrorHandler.handle(ReportPortalErrorHandler.java:59)
        at com.epam.reportportal.restendpoint.http.HttpClientRestEndpoint$1.subscribe(HttpClientRestEndpoint.java:517)
        at io.reactivex.internal.operators.maybe.MaybeCreate.subscribeActual(MaybeCreate.java:45)
        at io.reactivex.Maybe.subscribe(Maybe.java:4096)
        at io.reactivex.internal.operators.maybe.MaybeCache.subscribeActual(MaybeCache.java:77)
        at io.reactivex.Maybe.subscribe(Maybe.java:4096)
        at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeTask.run(MaybeSubscribeOn.java:54)
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38)
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
com.epam.reportportal.exception.ReportPortalException: Report Portal returned error
Status code: 406
Status message: Not Acceptable
Error Message: Start time of child ['Mon Mar 25 08:03:03 UTC 2019'] item should be same or later than start time ['Mon Mar 25 08:03:04 UTC 2019'] of the parent item/launch '5c988b39af66c600012a0f39'
Error Type: CHILD_START_TIME_EARLIER_THAN_PARENT

        at com.epam.reportportal.service.ReportPortalErrorHandler.handleError(ReportPortalErrorHandler.java:83)
        at com.epam.reportportal.service.ReportPortalErrorHandler.handle(ReportPortalErrorHandler.java:59)
        at com.epam.reportportal.restendpoint.http.HttpClientRestEndpoint$1.subscribe(HttpClientRestEndpoint.java:517)
        at io.reactivex.internal.operators.maybe.MaybeCreate.subscribeActual(MaybeCreate.java:45)
        at io.reactivex.Maybe.subscribe(Maybe.java:4096)
        at io.reactivex.internal.operators.maybe.MaybeCache.subscribeActual(MaybeCache.java:77)
        at io.reactivex.Maybe.subscribe(Maybe.java:4096)
        at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeTask.run(MaybeSubscribeOn.java:54)
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38)
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

Failed scenarios:
src/test/resources/scenario_outline_test.feature:11 # Dummy scenario
```
In order to reproduce the issue, please reuse the same serenity-with-cucumber-scenario-outlines.zip

with updated pom.xml: 

```java
<serenity.reportportal.version>1.4.0-SNAPSHOT</serenity.reportportal.version>

<repositories>
    <repository>
      <id>snapshots-repo</id>
      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
      <releases><enabled>true</enabled></releases>
      <snapshots><enabled>true</enabled></snapshots>
    </repository>
  </repositories>
```

**Comment by Invictum** [comment]
@nstbushneva nice catch!
Fix is uploaded to the same artifact name.

**Comment by Invictum** [comment]
Fixed. Will be available in version 1.4.0+

Parametrized tests may have equal names in RP, but always have a different set of parameters.

**Names**

<img width="1160" alt="Screenshot 2019-04-03 10 03 13" src="https://user-images.githubusercontent.com/4653335/55489562-c7454700-55f7-11e9-8ab5-b9b7e9b8f6a7.png">

**Parameters**

<img width="794" alt="Screenshot 2019-04-03 10 04 33" src="https://user-images.githubusercontent.com/4653335/55489639-eb088d00-55f7-11e9-8eaf-d7b9a2654f56.png">
